### PR TITLE
Remove option to use MQTT 3

### PR DESCRIPTION
**Pull request by underhood**
##### Summary
As we use MQTT 5 for multiple months and it is deemed stable enough now it is time to remove the feature to use MQTT 3. This allows also cleanup in mqtt_websockets library after the next stable release (with this PR in).

Reason: There is no real benefit to using MQTT 3 over MQTT 5 for user. Actually MQTT 3 is more problematic. We allowed MQTT 3 usage until now only as fallback until confidence in stability of new MQTT 5 implementation was high enough.

##### Test Plan
- MQTT 5 is always used
- nothing explodes

##### Additional Information

<details> <summary>For users: How does this change affect me?</summary>
  
</details>

### CODE REVIEW

File: aclk/aclk.c [hunk 1]
```diff
@@ -461,9 +460,9 @@ static int aclk_block_till_recon_allowed() {
  */
 static int aclk_get_transport_idx(aclk_env_t *env) {
     for (size_t i = 0; i < env->transport_count; i++) {
-        // currently we support only MQTT 3
+        // currently we support only MQTT 5
         // therefore select first transport that matches
-        if (env->transports[i]->type == ACLK_TRP_MQTT_3_1_1) {
+        if (env->transports[i]->type == ACLK_TRP_MQTT_5) {
```
Comment (stelfrag) [inline aclk/aclk.c:465]: This fails since only ACLK_TRP_MQTT_3_1_1 is received from the cloud

Comment (underhood) [reply aclk/aclk.c:465]: Cloud change was deployed on production therefore I removed the `[WIP]` label

File: aclk/aclk.h [hunk 2]
```diff
@@ -14,7 +14,7 @@
 #endif /* ENABLE_ACLK */
 
 extern int aclk_connected;
-extern int use_mqtt_5;
+extern int aclk_alert_reloaded;
```
Comment (MrZammler) [inline aclk/aclk.h:17]: Hi Timo! I think we don't need this... ?

File: aclk/aclk_tx_msgs.c [hunk 3]
```diff
@@ -87,21 +84,7 @@ static int aclk_send_message_with_bin_payload(mqtt_wss_client client, json_objec
...
-        if (rc == MQTT_WSS_ERR_BLOCK_TIMEOUT) {
-            error("Timeout sending binpacked message");
-            return HTTP_RESP_BACKEND_FETCH_FAILED;
-        }
-        if (rc == MQTT_WSS_ERR_TX_BUF_TOO_SMALL) {
-            error("Message is bigger than allowed maximum");
-            return HTTP_RESP_FORBIDDEN;
-        }
-    }
+    mqtt_wss_publish5(client, (char*)topic, NULL, (char*)(payload_len ? full_msg : str), (payload_len ? &freez_aclk_publish5b : &json_object_put_wrapper), len, MQTT_WSS_PUB_QOS1, &packet_id);
```
Comment (MrZammler) [inline aclk/aclk_tx_msgs.c:87]: Just a small warning here about `unused variable ‘rc’`

**Comment by MrZammler** [comment]
Running it locally, seems fine!

One small thing maybe, I do get the warning `client_id provided is longer than 23 bytes, server might not allow that [MQTT-3.1.3-5]` is this needed any more?

**Comment by underhood** [comment]
> Running it locally, seems fine!
> 
> One small thing maybe, I do get the warning `client_id provided is longer than 23 bytes, server might not allow that [MQTT-3.1.3-5]` is this needed any more?

This is correct warning. According to MQTT v5 official specification server is not required to accept client_id longer than 23 bytes. Our claim_id is longer.

File: aclk/aclk.h (hunk 2 above)
Comment (underhood) [reply aclk/aclk.h]: Thanks, rebase mistake

File: aclk/aclk_tx_msgs.c (hunk 3 above)
Comment (underhood) [reply aclk/aclk_tx_msgs.c:87]: thx, removed

**Review by thiagoftsm — APPROVED** [review]
Everything is working as expected, LGTM!

**Pull request merged**